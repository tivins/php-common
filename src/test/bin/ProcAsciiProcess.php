<?php

use Tivins\Core\Proc\AsciiProcess;
use Tivins\Core\Proc\Command;
use Tivins\Core\System\File;

require 'vendor/autoload.php';

// Create a simple PHP file for example.
$tmpFile = tempnam('/tmp', 'test');
File::save($tmpFile, '<?' . 'php' . "\n" . 'echo "Hello"; sleep(1); echo "Word\n";');

// Build cast with command 'php /path/to/file.php'
$cast = AsciiProcess::buildCast(new Command('php', $tmpFile));
echo $cast . PHP_EOL;

// Remove temp file.
File::delete($tmpFile);

