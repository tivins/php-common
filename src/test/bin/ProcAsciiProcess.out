{"version":2,"width":80,"height":10,"timestamp":1674502452,"env":{"SHELL":"\/bin\/bash","TERM":"xterm-256color"}}
[0,"o","\u0007\u001b[01;32mme@hostname\u001b[00m:\u001b[01;34m\/\u001b[00m$ run.php\r\n"]
[0.03485393524169922,"o","Hello"]
[1.0348260402679443,"o","Word\r\n"]
[1.0506720542907715,"o","\u0007\u001b[01;32mme@hostname\u001b[00m:\u001b[01;34m\/\u001b[00m$ "]
